homeassistant/components/fritzbox_callmonitor/__init__.py

```diff
@@ -1 +1,52 @@
-"""The fritzbox_callmonitor component."""
+"""The fritzbox_callmonitor integration."""
+import asyncio
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+from .base import FritzBoxPhonebook
+from .const import CONF_PHONEBOOK, CONF_PREFIXES, DOMAIN, PLATFORMS
+
+
+async def async_setup(hass, config):
+    """Set up the fritzbox_callmonitor integration."""
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up the fritzbox_callmonitor platforms."""
+    phonebook = FritzBoxPhonebook(
+        host=entry.data[CONF_HOST],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        phonebook_id=entry.data[CONF_PHONEBOOK],
+        prefixes=entry.data[CONF_PREFIXES],
+    )
+    await hass.async_add_executor_job(phonebook.init_phonebook)
```

Comment:
Since this method checks if the phonebook is valid, can we catch if the auth no longer works? 

And if we can't connect, raise a `ConfigEntryNotReady`

Reply:
Have added a check for all exceptions that could occur and raise `ConfigEntryNotReady` if one occurs.